main.c

```diff
@@ -25,6 +25,7 @@
  */
 #include "scrypt_platform.h"
 
+#include <math.h>
```

Comment:
I added this for INFINITY, used (at the moment) for `-t maxtime`.  I'm not certain if you want an upper bound on that -- sure, if they wanted to not time out, they'd just omit the `-t` parameter, but the original code didn't impose an upper bound.